prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt

```diff
@@ -203,6 +204,56 @@ class DeliveryFunction(
         }
     }
 
+    /**
+     * Get a sortable list of facilities for multiple deliveries
+     *
+     * @param request HTML request body.
+     * @return JSON of the facility list or errors.
+     */
+    @FunctionName("getDeliveryFacilitiesBulk")
+    fun getDeliveryFacilitiesBulk(
+        @HttpTrigger(
+            name = "getDeliveryFacilitiesBulk",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "waters/report/facilities"
+        ) request: HttpRequestMessage<String?>,
+    ): HttpResponseMessage {
+        try {
+            return run {
+                val body = request.body ?: return HttpUtilities.bad(request, "Body must be provided")
+                val reportIds = body.split(",").mapNotNull { ReportId.fromString(it) }
+
+                val facilities = deliveryFacade.findDeliveryFacilitiesBulk(
```

Comment:
Does this endpoint also need to be concerned that all the requested report ids are ultimately for the same receiver?

Reply:
We should be fine now, with the receiver being part of the required parameters.